daisy_workflows/image_import/import_image.sh

```diff
@@ -48,24 +54,50 @@ function resizeDisk() {
   fi
 }
 
-# Mount GCS bucket containing the disk image.
-mkdir -p /gcs/${SOURCEBUCKET}
-gcsfuse --implicit-dirs ${SOURCEBUCKET} /gcs/${SOURCEBUCKET}
+function copyImageToScratchDisk() {
+  local sourceSizeGigabytes="$((1 + (SOURCE_SIZE_BYTES - 1) / BYTES_1GB))"
+  local scratchDiskSizeGigabytes=sourceSizeGigabytes
+  # We allocate double capacity for OVA, which would
+  # require making an additional copy of its enclosed VMDK.
+  if [[ "${SOURCE_FILE_TYPE}" == "ova" ]]; then
+     scratchDiskSizeGigabytes=$((sourceSizeGigabytes * 2))
+  fi
+
+  # This disk is initially created with 10GB of space.
+  # Enlarge it if that's insufficient to hold the input image.
+  if [[ ${scratchDiskSizeGigabytes} -gt 10 ]]; then
+    resizeDisk "${SCRATCH_DISK_NAME}" "${scratchDiskSizeGigabytes}" "${ZONE}"
+  fi
+
+  local scratchPath="/daisy-scratch/$(dirname ${SOURCEPATH})/"
+  mkdir -p "${scratchPath}"
+  # /dev/sdb is used since the scratch disk is the second
+  # disk that's attached in import_disk.wf.json.
+  mkfs.ext4 /dev/sdb
+  mount /dev/sdb "${scratchPath}"
+  if [[ $? -ne 0 ]]; then
+    echo "ImportFailed: Failed to prepare scratch disk."
+  fi
+  gsutil cp "${SOURCEURL}" "${scratchPath}"
+  echo "Import: Copied image from ${SOURCEURL} to ${scratchPath}"
+}
+
+copyImageToScratchDisk
```

Comment:
why define a function only to immediately call it once?

Reply:
I agree that since the subsequent code is outside of a function this looks a bit awkward. The benefits are limiting the scope of variables inside the function, as well as putting a name around a block of code. I think the benefits are worth it. But I could be convinced otherwise!

If I weren't so averse to refactoring hard-to-test bash scripts, I'd break the subsequent block of code into a couple of functions and then the top-level script would be: `copyImageToScratchDisk && checkAndInflateOva && writeImageToDisk`

Reply:
makes sense. i think you'd be supported in rewriting a script of this length if you wanted, but up to you.

Reply:
@hopkiw This is a interesting topic actually. I faced similar situations - I want to write a sub script, but there isn't a good way to copy it to the machine (not like startup script). I will have to download it before calling it. Do you have a better idea?